icu4c/source/tools/genren/genren.pl

```diff
@@ -154,7 +154,9 @@
             chop $qtype;
         } elsif ($mode =~ /Mach-O/) {
             ($address, $type, $_) = split(/ /);
-            if(/^_(.*)$/) {
+            if (/^(.*)\(/) {
```

Comment:
Like below, a comment would be nice for what we do here and why. Maybe
// On macOS, C++ function symbols include their full signature, with (arguments in parens). Remove those.